### PR TITLE
Don't require the code be placed inside GOPATH.

### PR DESCRIPTION
**Pull request by metlos**
## Description
Hack around the fact that openapi-gen requires the codebase be checked out in a GOPATH.

## Checks
1. Did you run `make generate` target? **yes**

2. Did `make generate` change anything in other projects (host-operator, member-operator)? **no, no changes to the CRDs were made in the PR** 

3. In case of **new** CRD, did you the following? **no, no changes to the CRDs were made in the PR**
    - make/generate.mk in this repository
    - `resources/setup/roles/host.yaml` in the sandbox-sre repository
    - `PROJECT` file: https://github.com/codeready-toolchain/host-operator/blob/master/PROJECT
    - `CSV` file: https://github.com/codeready-toolchain/host-operator/blob/master/config/manifests/bases/host-operator.clusterserviceversion.yaml

4. In case other projects are changed, please provides PR links.
    - host-operator: https://github.com/codeready-toolchain/host-operator/pull/#
    - member-operator: https://github.com/codeready-toolchain/member-operator/pull/#

### CODE REVIEW

File: make/generate.mk [hunk 1]
```diff
@@ -18,15 +18,32 @@ generate: generate-deepcopy-and-crds generate-openapi dispatch-crds copy-reg-ser
...
 .PHONY: generate-openapi
 generate-openapi: openapi-gen
 	@echo "re-generating the openapi go file..."
-	$(Q)$(OPENAPI_GEN) --input-dirs ./api/$(API_VERSION)/ \
+	@## openapi-gen requires the GOPATH env var be set and the codebase be present within it.
+	@## Let's not require $GOPATH be set up in the user's environment and the checkout be
+	@## placed in it.
+	@## Instead, fake it locally.
+	mkdir -p $(PROJECT_DIR)/.fake-gopath
+	@mkdir -p $(PROJECT_DIR)/.fake-gopath/src/github.com/codeready-toolchain
```
Comment (MatousJobanek) [inline make/generate.mk:31]: it's just a detail, how about storing `src/github.com/codeready-toolchain` as a variable?

File: make/generate.mk [hunk 2]
```diff
@@ -18,15 +18,32 @@ generate: generate-deepcopy-and-crds generate-openapi dispatch-crds copy-reg-ser
...
+	@## link the packages from the real GOPATH
+	@## ("go env GOPATH" returns the correct value even if the env var is not set)
+	@cd $(PROJECT_DIR)/.fake-gopath && ln -s `go env GOPATH`/pkg || true
```
Comment (MatousJobanek) [inline make/generate.mk:34]: out of curiosity, why do we need `|| true`?

File: make/generate.mk [hunk 3]
```diff
@@ -18,15 +18,32 @@ generate: generate-deepcopy-and-crds generate-openapi dispatch-crds copy-reg-ser
...
+	@## link our codebase to the appropriate place in the fake GOPATH
+	@cd $(PROJECT_DIR)/.fake-gopath/src/github.com/codeready-toolchain && ln -s ../../../.. api
+	@## run openapi-gen from within the fake GOPATH (otherwise the package paths would be relative
+	@#e and function names would be different)
+	export GOPATH=$(PROJECT_DIR)/.fake-gopath \
+	&& cd $(PROJECT_DIR)/.fake-gopath/src/github.com/codeready-toolchain/api \
+	&& $(Q)$(OPENAPI_GEN) --input-dirs ./api/$(API_VERSION)/ \
```
Comment (MatousJobanek) [inline make/generate.mk:41]: I guess that we don't need to export the variable, right? this should be enough:
```suggestion
	&& cd $(PROJECT_DIR)/.fake-gopath/src/github.com/codeready-toolchain/api \
	&& GOPATH=$(PROJECT_DIR)/.fake-gopath $(Q)$(OPENAPI_GEN) --input-dirs ./api/$(API_VERSION)/ \
```

Comment (MatousJobanek) [reply make/generate.mk:41]: btw, we can also drop the `$(Q)` - no clue why it is there
```suggestion
	&& cd $(PROJECT_DIR)/.fake-gopath/src/github.com/codeready-toolchain/api \
	&& GOPATH=$(PROJECT_DIR)/.fake-gopath $(OPENAPI_GEN) --input-dirs ./api/$(API_VERSION)/ \
```

**Review by MatousJobanek — COMMENTED** [review]
Thanks a lot for fixing this :pray: 
I have a few questions/comments

File: make/generate.mk (hunk 2 above)
Comment (metlos) [reply make/generate.mk:34]: Heh, `ln` works differently than I thought...

**Review by MatousJobanek — APPROVED** [review]
Cool, thanks for fixing it

**Pull request merged**